xds/src/main/java/io/grpc/xds/ServerXdsClient.java

```diff
@@ -99,12 +110,10 @@ private void updateNodeMetadataForListenerRequest(int port) {
     if (node.getMetadata() != null) {
       newMetadata.putAll(node.getMetadata());
     }
-    newMetadata.put("TRAFFICDIRECTOR_PROXYLESS", "1");
-    // TODO(sanjaypujare): eliminate usage of listening_addresses.
-    EnvoyProtoData.Address listeningAddress =
-        new EnvoyProtoData.Address("0.0.0.0", port);
-    node =
-        node.toBuilder().setMetadata(newMetadata).addListeningAddresses(listeningAddress).build();
+    newMetadata.put("TRAFFICDIRECTOR_INBOUND_INTERCEPTION_PORT", "15001");
```

Comment:
15001 is fixed? Do you have a design for that?

Reply:
This is the "fallback" case where we simulate the Envoy-like xDS API to talk to the TD and this is going to go away so there is no point in having a design. Once the `newServerApi` mode is shown to be tested in production and working then the fallback/legacy code will be removed